### PR TITLE
Basic implementation of Keyboard Shortcut Editor [studio]

### LINKED ISSUE #422: Feature request: Shortcut editing in preferences
**Synfig version & platform**: Synfig 1.2.0 Win10
(we're more likely to help if you're using latest development builds)

**Issue description**:
Feature request:
Chance to edit shortcuts in preferences. Also which mouse/pen buttons used.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/49158965-feature-request-shortcut-editing-in-preferences?utm_campaign=plugin&utm_content=tracker%2F321991&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F321991&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by rodolforg**
Some other features could be done, for example:
- Own button for 'restore default shortcuts only' (instead of together with all other Synfig Studio settings)
- Save and load specific/custom accel map file (it always read/overwrite [config-folder]/accelrc)
- List all available actions (currently it lists only those with a default shortcut set in app.cpp)
- Group actions in a Tree view structure (but I'm lazy for that)
- List with better (simpler) names and maybe tooltips: right now it's the action full path. However, it would require a proper registration on every case, and this PR is just a quick implementation
- Warn user if a shortcut is used for another action

fix #422

### CODE REVIEW

File: synfig-studio/src/gui/dialogs/dialog_setup.cpp [hunk 1]
```diff
@@ -447,6 +457,69 @@ dragging the handle to the left bottom part of your 2D space."));
...
+	pi.grid->attach(*scroll, 0, 1, 1, 1);
+}
+
+void
+Dialog_Setup::on_accel_edited(const Glib::ustring& path_string, guint accel_key, Gdk::ModifierType accel_mods, guint hardware_keycode)
+{
+	auto iter = treeview_accels->get_model()->get_iter(path_string);
+	iter->set_value(SHORTCUT_COLUMN_ID_ACTION_KEY, accel_key);
+	iter->set_value(SHORTCUT_COLUMN_ID_ACTION_MODS, accel_mods);
+	std::string action_name;
```
Comment (ice0) [inline synfig-studio/src/gui/dialogs/dialog_setup.cpp:513]: Why is that needed?
```
std::string action_name;
iter->get_value(SHORTCUT_COLUMN_ID_ACTION_NAME, action_name);
```
Looks like it do nothing.

**Comment by ice0** [comment]
I suggest to add a little help to that page. Something like:
```
To change a shortcut, select it, double-click shortcut and then type the new keys.
```

Also I didn't figure out how to clear the shortcut.

**Comment by rodolforg** [comment]
> Also I didn't figure out how to clear the shortcut.

Backspace. Yeah, that should be explained too lol (I used the Gtk implementation –Gtk::CellRendererAccel– and i had a hard time trying to figure it out)

File: synfig-studio/src/gui/dialogs/dialog_setup.cpp (hunk 1 above)
Comment (rodolforg) [reply synfig-studio/src/gui/dialogs/dialog_setup.cpp:513]: Of course it does: it shows everyone I didn't properly clean my code ;)

File: synfig-studio/src/gui/dialogs/dialog_setup.cpp [hunk 2]
```diff
@@ -447,6 +457,75 @@ dragging the handle to the left bottom part of your 2D space."));
 	
 }
 
+void
+Dialog_Setup::create_shortcuts_page(Dialog_Template::PageInfo pi)
+{
+	int row = 1;
+
+	Gtk::Label *label = manage(new Gtk::Label(_("To change a shortcut, double-click it and then type the new keys.\n"
```
Comment (ice0) [inline synfig-studio/src/gui/dialogs/dialog_setup.cpp:465]: Maybe remove new line after first message? As for me it looks better because there no free space on the right side, and message is more compact.

Screenshot_149

Screenshot_150

Comment (rodolforg) [reply synfig-studio/src/gui/dialogs/dialog_setup.cpp]: Done

**Review by ice0 — APPROVED** [review]

**Comment by ice0** [comment]
Merged. Thank you!

**Comment by morevnaproject** [comment]
Great addition! I have added screenshot in the top post to use later in Release Notes. ^__^
